rust/examples/broadcast.rs

```diff
@@ -63,17 +63,28 @@ async fn main() -> PyResult<()> {
         )
         .map_err(|e| anyhow!(e))?;
 
-    device.set_advertising_data(adv_data)?;
     device.power_on().await?;
 
-    println!("Advertising...");
-    device.start_advertising(true).await?;
+    if cli.extended {
+        println!("Starting extended advertisement...");
+        device.start_advertising_extended(adv_data).await?;
+    } else {
+        device.set_advertising_data(adv_data)?;
```

Comment:
why does the non-extended advertising need set advertisement data separately?

Reply:
Non-extended advertising sets advertisement data separately to mirror the python behavior.

In adding extended advertising, I don't want to interfere with how `advertisement_data` is used for legacy use cases, so it takes advertisement data as a parameter.

Reply:
ah gotcha, perhaps `set_advertising_data` should be renamed to `set_legacy_advertising_data` then?

Reply:
This raises an interesting question.

`set_advertising_data` literally sets a new value for the property called `advertising_data`.

So, is it better to (1) rename it to `set_legacy_advertising_data` and document that it sets the property called `advertising_data`, or (2) to call it `set_advertising_data` and document that it gets used for legacy advertisements?

Since extended advertising is gated as an "unstable" feature, I'm inclined toward (2), but I want to know what you think.

Reply:
Really, both have their merits. 1 is a better API overall, but 2 will cause less cognitive overhead for people switching from using the Python API.

Reply:
Unless @barbibulle has some opinion on it, I'm probably going to stick with how it is currently. It may be confusing to some Rust users, but only if you opt-in for the unstable extended API, in which case you are expected to know what you're signing up for.

Plus, when extended is eventually added to `bumble` core, this API may disappear anyway.